gitattributes.go

```diff
@@ -0,0 +1,251 @@
+package enry
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	"gopkg.in/src-d/enry.v1/data"
+)
+
+type attrType int
+
+const (
+	vendor attrType = iota
+	documentation
+	generated
+	language
+)
+
+const _attrType_name = "vendordocumentationgeneratedlanguage"
+
+var _attrType_index = [...]uint8{0, 6, 19, 28, 36}
+
+func (i attrType) String() string {
+	if i < 0 || i >= attrType(len(_attrType_index)-1) {
+		return fmt.Sprintf("attrType(%d)", i)
+	}
+	return _attrType_name[_attrType_index[i]:_attrType_index[i+1]]
+}
+
+type boolAttribute struct {
+	kind       attrType
+	matchers   []string
+	attributes map[string]bool
+}
+
+type regExpAttribute struct {
+	kind       attrType
+	matchers   []string
+	attributes map[*regexp.Regexp]string
+}
+
+// GitAttributes is a struct that contains two maps, boolAttributes contains all the attributes that works like a boolean condition,
+// regExpAttributes contains all the attributes that match a regExp to choose if an attribute is applied or not
+type GitAttributes struct {
+	boolAttributes   map[attrType]boolAttribute
+	regExpAttributes map[attrType]regExpAttribute
+}
+
+type overrideError struct {
+	attribute attrType
+	path      string
+}
+
+func (e *overrideError) Error() string {
```

Comment:
Well,  if an error is found nothing happens when the LoadAttributes function ends, it returns a slice of warnings that you can ignore, I'm only using this for traceability and testing purposes